scripts/pancancer.py

```diff
@@ -0,0 +1,341 @@
+        if subparser_name=='list':
+            workflow_list = WorkflowLister.get_workflow_names()
+            print ('Available workflows are:')
+            print(workflow_list)
+
+        elif subparser_name=='config':
+            workflow_name=vars(parsed_args)['workflow_name']
+            if workflow_name in WorkflowLister.get_workflow_names():
+                # TODO: Actually generate the correct INI file!
+                print('generating default INI in /home/ubuntu/ini-dir/'+workflow_name+'.ini for workflow '+workflow_name)
+            else:
+                print ('Sorry, but '+workflow_name+' is not a valid workflow name. Please use the command \'workflows list\' to see a list of available workflows.')
+        else:
+            print ('The available subcommands are \'list\' and \'config\'.')
+
+###
+
+class DaemonCommand(Command):
+    "Parent class for commands that start/stop daemons"
+    service_name=''
+    log = logging.getLogger(__name__)
+
+    def _do_start(self,start_cmd):
+        self.log.debug (start_cmd)
+        p = subprocess.Popen(start_cmd.split(' '))
+        # Wait a moment... if we get ANY response, it means that the process did not start, probably because it's already been started.
+        time.sleep(2)
+        p.poll()
+        if p.returncode==1:
+            print('The '+self.service_name+' process might already be running.')
+        else:
+            print('The service has been started, check the log files for more detail.')
+
+    def _do_stop(self,stop_cmd):
+        self.log.debug (stop_cmd)
+        p = subprocess.Popen(stop_cmd.split(' '))
+        # wait a moment to get the result of the kill command.
+        time.sleep(2)
+        p.poll()
+        if p.returncode==0:
+            print('The '+self.service_name+' process has been stopped.')
+        elif p.returncode==1:
+            print('The '+self.service_name+' process does not appear to have been running, nothing has been stopped.')
+
+    def get_parser(self,prog_name):
+        parser = super(DaemonCommand,self).get_parser(prog_name)
+        subparser = parser.add_subparsers(title='subcommands',help='coordinator subcommands: start, stop, restart',dest='service_state')
+        subparser.required=True
+        subparser.add_parser('start',help='Start the '+self.service_name+' service')
+        subparser.add_parser('stop',help='Stop the '+self.service_name+' service')
+        subparser.add_parser('restart',help='Stop the '+self.service_name+' service')
+        return parser
+
+    def take_action(self, parsed_args):
+        subparser_name=vars(parsed_args)['service_state']
+        self.log.debug('working on service: '+self.service_name)
+        self.log.debug('subparser: %s',subparser_name)
+        start_cmd='flock -n /tmp/arch3_'+self.service_name+'.pid ' +self.service_name+' --config /home/ubuntu/arch3/config/masterConfig.ini --endless'
+        stop_cmd='pkill -f '+self.service_name
```

Comment:
Flock doesn't produce an actual pid file, the ".pid" is just a part of its name, but I'll see about capturing the pid properly and using that to kill the process.